lib/modules/manager/poetry/extract.ts

```diff
@@ -124,7 +138,11 @@ export async function extractPackageFile(
     ...extractFromSection(pyprojectfile, 'dependencies', lockfileMapping),
     ...extractFromSection(pyprojectfile, 'dev-dependencies', lockfileMapping),
     ...extractFromSection(pyprojectfile, 'extras', lockfileMapping),
+    ...Object.keys(pyprojectfile.tool?.poetry?.group ?? []).flatMap((group) =>
+      extractFromSection(pyprojectfile, new PoetryGroup(group), lockfileMapping)
```

Comment:
Would it be beneficial to extract the group's name, e.g. to put in depType?

Reply:
I wasn't sure how Renovate is using `depType`. I believe it's used for package rules (e.g., for applying labels or setting package rules on the type of dependency ?). If so yes, I believe having such information would be useful.

I added https://github.com/renovatebot/renovate/pull/17964/commits/57263846ef919a85006eda1170f4e2e9d5ed1779, but it feels a bit weird to me to have `dependencies`, `dev-dependencies`, and the raw group name for the groups. Especially since `[tool.poetry.dev-dependencies]` is now deprecated, and dev dependencies should now be defined under `[tool.poetry.group.dev.depedencies]`, so users migrating will now have `dev` instead of `dev-dependencies` for `depType`.